docs/okeps/okep-5296-bgp.md

```diff
@@ -0,0 +1,663 @@
+When omitted, RouteAdvertisements will be applied to the default cluster network. If a selector is used that selects the
+default cluster network, it will be enabled for any namespace using the cluster default network. Additionally,
+advertisements may be limited to specific nodes via the nodeSelector.
+
+A networkSelector may select more than just the cluster default network, including user-defined networks. In such a case
+the network subnets will be checked by OVN-Kubernetes to determine if there is any overlap of the IP subnets. If so, an
+error status will be reported to the CRD and no BGP configuration will be done by OVN-Kubernetes.
+
+Multiple CRs may not select the same network. If this happens OVN-Kubernetes will report an error to the
+RouteAdvertisements CR and will refuse to apply the config.
+
+The CRD will support enabling advertisements for pod subnet and egress IPs. Note, MetalLB still handles advertising
+LoadBalancer IP so there is no conflict of responsibilities here. When the pod network is set to be advertised, there is
+no longer a need to SNAT pod IPs for this network to the node IP. Therefore, when pod network advertisements are enabled,
+the traffic from these pods will no longer be SNAT'ed on egress.
+
+The "targetVRF" key is used to determine which VRF the routes should be advertised in. The default value is "default", in
+which case routes will be leaked into the default VRF. One use case for this would be when a user wants to define a
+network with a specific IP addressing scheme, and then wants to advertise the pod IPs into the provider BGP network without
+VPN. Note that by using route leaking with a user-defined network, the network is no longer fully
+isolated, as now any other networks also leaked or attached to that VRF may reach this user-defined network. If a user
+attempts to leak routes into a targetVRF for a user-defined network whose IP subnet would collide with another,
+OVN-Kubernetes will report an error to the RouteAdvertisement status. Alternatively, a user may specify the value "auto",
+which in which case OVN-Kubernetes will advertise routes in the VRF that corresponds to the selected network. Any other
+values other than "auto" or "default" will result in an error. There is no support at this time for routing leaking into
+any other VRF.
+
+The frrConfigurationSelector is used in order to determine which FRRConfiguration CR to use for building the OVN-Kubernetes
+driven FRRConfiguration. OVN-Kubernetes needs to leverage a pre-existing FRRConfiguration to be able to find required
+pieces of configuration like BGP peering, etc. If more than one FRRConfiguration is found matching the selector, then
+an error will be propagated to the RouteAdvertisements CR and no configuration shall be done.
+
+#### No Tunnel/Overlay Mode
+
+Changes will be made to the CUDN and UDN CRDs in order to specify the method of transport for east/west traffic. A new
```

Comment:
do we keep this and link to no-overlay enhancement PR for more details? https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5289